spec/erb_lint/linters/content_style_spec.rb

```diff
@@ -0,0 +1,629 @@
+
+require 'spec_helper'
+
+describe ERBLint::Linter::ContentStyle do
+  let(:linter_config) do
+    {
+      'rule_set' => rule_set,
+      'addendum' => 'Questions? Contact Lintercorp Product Content at product-content@lintercorp.com.'
+    }
+  end
+
+  let(:linter) { described_class.new(linter_config) }
+
+  subject(:linter_errors) { linter.lint_file(ERBLint::Parser.parse(file)) }
+
+  context 'when the rule set is empty' do
+    let(:rule_set) { [] }
+
+    context 'when the file is empty' do
+      let(:file) { '' }
+
+      it 'does not report any errors' do
+        expect(linter_errors).to eq []
+      end
+    end
+  end
+
+  context 'when the rule set contains violations' do
+    context '- rule is case-insensitive
+    - file contains violation with different case from suggestion (`Drop down`)
+    - file contains violation with same case as suggestion (`dropdown`)
+    - file contains suggestion (`drop-down`)' do
+      violation_set_1 = ['dropdown', 'drop down']
+      suggestion_1 = 'drop-down'
+      case_insensitive_1 = true
+
+      let(:rule_set) do
+        [
+          {
+            'violation' => violation_set_1,
+            'suggestion' => suggestion_1,
+            'case_insensitive' => case_insensitive_1
+          }
+        ]
+      end
+
+      let(:file) { <<~FILE }
+        <p>Tune in, turn on, and drop-down out! And check out the Drop down and dropdown menu too.</p>
+
+      FILE
+
+      it 'reports 2 errors' do
+        expect(linter_errors.size).to eq 2
+      end
+
+      it 'reports errors for `Drop down` and `dropdown` and suggests `drop-down`' do
+        expect(linter_errors[0][:message]).to include 'Don\'t use `dropdown`'
+        expect(linter_errors[0][:message]).to include 'Do use `drop-down`'
+        expect(linter_errors[1][:message]).to include 'Don\'t use `drop down`'
+        expect(linter_errors[1][:message]).to include 'Do use `drop-down`'
+      end
+    end
+
+    context '- suggestion is prefix + violation (`Lintercorp Help Center`)
+    - file contains suggestion (`Lintercorp Help Center`)
+    - file contains violation (`Help Center`)' do
+      violation_set_1 = ['Help Center', 'help center']
+      suggestion_1 = 'Lintercorp Help Center'
+
+      let(:rule_set) do
+        [
+          {
+            'violation' => violation_set_1,
+            'suggestion' => suggestion_1
+          }
+        ]
+      end
+
+      let(:file) { <<~FILE }
+        <p>Help! I need a Lintercorp Help Center. Not just any Help Center. Help!</p>
+
+      FILE
+
+      it 'reports 1 errors' do
+        expect(linter_errors.size).to eq 1
+      end
+
+      it 'reports error for `Help Center` and suggests `Lintercorp Help Center`' do
+        expect(linter_errors[0][:message]).to include 'Don\'t use `Help Center`'
+        expect(linter_errors[0][:message]).to include 'Do use `Lintercorp Help Center`'
+      end
+    end
+
+    context '- suggestion is prefix + violation (`Lintercorp theme store`)
```

Comment:
This test is containing way too much logic. You should be able to break this up into at least 2 tests. No need to have the File content have both `theme store` and `Theme Store` at once. One of them will already demonstrate having 1 violated rule inside of another.

Each unit test should be able to encapsulate what it is testing in one sentence (the `context` descriptions).